R/object_name_linters.R

```diff
@@ -24,6 +24,27 @@ object_name_xpath <- local({
 #' Check that object names conform to a naming style.
 #' The default naming styles are "snake_case" and "symbols".
 #'
+#' Note when used in a package, in order to ignore objects imported
+#'   from other namespaces, this linter will attempt [getNamespaceExports()]
+#'   whenever an `import(PKG)` statement is found in your NAMESPACE file.
+#'   If this fails (e.g., the package is not yet installed), the linter
+#'   won't be able to ignore such usages.
+#'
+#' Suppose, for example, you have `import(upstream)` in your NAMESPACE,
+#'   which makes available its exported function
+#'   `a_really_quite_long_function_name` that you then use in your package.
+#'   Then, if `upstream` is not installed when this linter runs, a lint
+#'   will be thrown on this object (even though you don't "own" its name).
+#'
+#' There are three options to get lintr to work correctly:
+#'   (1) install the package so that it's available in
+#'   the session where this linter is running; (2) always namespace-qualify
+#'   usages, because this linter ignores names in `pkg::foo()` form; and
+#'   (3) use `importFrom(pkg, foo)` instead of a blanket `import(pkg)` in your
+#'   NAMESPACE, because this linter takes these imported names as given;
+#'   and of course (3) install the package so that it's available in
+#'   the session where this linter is running.
```

Comment:
(3) appears twice now.
Also, what about using a md enumeration?

1. install the package so that it's available to this linter while it's running.
2. only use namespace-qualified functions, e.g. `pkg::foo()`, because they are ignored by this linter.
3. only use selective imports (`importFrom(pkg, foo)`) instead of wildcard imports (`import(pkg)`) in your `NAMESPACE`, because these are always assumed to be available by this linter.

Reply:
NVM, I think this list is wrong anyway. Resolving.